CMakeLists.txt

```diff
@@ -4,6 +4,11 @@ cmake_minimum_required(VERSION 3.5)
 set(CMAKE_MACOSX_RPATH 1)
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
 
+if (APPLE)
+    set(CMAKE_INSTALL_RPATH "@loader_path")
+elseif (UNIX)
+    set(CMAKE_INSTALL_RPATH "$ORIGIN")
+endif()
```

Comment:
This is to make sure the relative linking works. Apple uses `@loader_path` for libraries and `@executable_path` for executables, while Linux uses `$ORIGIN`.

Reply:
rpaths also confuse me so could be wrong here